protocols/kad/src/behaviour.rs

```diff
@@ -214,13 +214,17 @@ impl Default for KademliaConfig {
 }
 
 impl KademliaConfig {
-    /// Sets a custom protocol name.
+    /// Sets custom protocol names.
     ///
     /// Kademlia nodes only communicate with other nodes using the same protocol
-    /// name. Using a custom name therefore allows to segregate the DHT from
+    /// name. Using custom name(s) therefore allows to segregate the DHT from
     /// others, if that is desired.
-    pub fn set_protocol_name(&mut self, name: impl Into<Cow<'static, [u8]>>) -> &mut Self {
```

Comment:
Sorry, contrary to what I suggested on https://github.com/libp2p/rust-libp2p/issues/2837, how about introducing `set_protocol_nameS` here while still supporting `set_protocol_namE`.

For now we would add a [deprecation warning](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute) on `set_protocol_namE` but have it call `set_protocol_nameS` instead.

Once released, we can remove `set_protocol_namE` in the subsequent release.

Reply:
Done.